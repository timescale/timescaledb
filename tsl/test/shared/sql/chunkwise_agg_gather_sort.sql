-- This file and its contents are licensed under the Timescale License.
-- Please see the included NOTICE for copyright information and
-- LICENSE-TIMESCALE for a copy of the license.

-- Exercise the GatherMerge -> Sort -> Append -> Partial Aggregate plan.

\set prefix 'explain (costs off, timing off, summary off)'

set parallel_setup_cost = 0;
set parallel_tuple_cost = 0;
set max_parallel_workers_per_gather = 2;
set parallel_leader_participation = off;

set enable_hashagg to off;

:prefix
select count(*) from metrics group by v0;

reset enable_hashagg;

:prefix
select count(*) from metrics group by v0;

reset parallel_setup_cost;
reset parallel_tuple_cost;
reset max_parallel_workers_per_gather;
reset parallel_leader_participation;
