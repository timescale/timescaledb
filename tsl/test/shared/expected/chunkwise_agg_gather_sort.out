-- This file and its contents are licensed under the Timescale License.
-- Please see the included NOTICE for copyright information and
-- LICENSE-TIMESCALE for a copy of the license.
-- Exercise the GatherMerge -> Sort -> Append -> Partial Aggregate plan.
\set prefix 'explain (costs off, timing off, summary off)'
set parallel_setup_cost = 0;
set parallel_tuple_cost = 0;
set max_parallel_workers_per_gather = 2;
set parallel_leader_participation = off;
set enable_hashagg to off;
:prefix
select count(*) from metrics group by v0;
QUERY PLAN
 Finalize GroupAggregate
   Group Key: _hyper_X_X_chunk.v0
   ->  Gather Merge
         Workers Planned: 2
         ->  Sort
               Sort Key: _hyper_X_X_chunk.v0
               ->  Parallel Append
                     ->  Partial GroupAggregate
                           Group Key: _hyper_X_X_chunk.v0
                           ->  Sort
                                 Sort Key: _hyper_X_X_chunk.v0
                                 ->  Seq Scan on _hyper_X_X_chunk
                     ->  Partial GroupAggregate
                           Group Key: _hyper_X_X_chunk.v0
                           ->  Sort
                                 Sort Key: _hyper_X_X_chunk.v0
                                 ->  Seq Scan on _hyper_X_X_chunk
                     ->  Partial GroupAggregate
                           Group Key: _hyper_X_X_chunk.v0
                           ->  Sort
                                 Sort Key: _hyper_X_X_chunk.v0
                                 ->  Seq Scan on _hyper_X_X_chunk
(22 rows)

reset enable_hashagg;
:prefix
select count(*) from metrics group by v0;
QUERY PLAN
 Finalize HashAggregate
   Group Key: _hyper_X_X_chunk.v0
   ->  Gather
         Workers Planned: 2
         ->  Parallel Append
               ->  Partial HashAggregate
                     Group Key: _hyper_X_X_chunk.v0
                     ->  Seq Scan on _hyper_X_X_chunk
               ->  Partial HashAggregate
                     Group Key: _hyper_X_X_chunk.v0
                     ->  Seq Scan on _hyper_X_X_chunk
               ->  Partial HashAggregate
                     Group Key: _hyper_X_X_chunk.v0
                     ->  Seq Scan on _hyper_X_X_chunk
(14 rows)

reset parallel_setup_cost;
reset parallel_tuple_cost;
reset max_parallel_workers_per_gather;
reset parallel_leader_participation;
